python/bind_operators.h

```diff
@@ -20,6 +20,28 @@ void bind_comparison(pybind11::class_<T, Ignored...> &c) {
       py::call_guard<py::gil_scoped_release>());
 }
 
+template <class Other, class T, class... Ignored>
+void bind_variable_comparison(pybind11::class_<T, Ignored...> &c) {
+  c.def(
+      "__eq__", [](T &a, Other &b) { return variable::equal(a, b); },
+      py::is_operator(), py::call_guard<py::gil_scoped_release>());
+  c.def(
+      "__ne__", [](T &a, Other &b) { return ~variable::equal(a, b); },
```

Comment:
Use `variable::not_equal` so we make use of potential optimizations there. If you negate here we are doing two operations instead of one.